storage/src/main/java/org/apache/kafka/server/log/remote/metadata/storage/ConsumerTask.java

```diff
@@ -64,302 +65,387 @@
 class ConsumerTask implements Runnable, Closeable {
     private static final Logger log = LoggerFactory.getLogger(ConsumerTask.class);
 
-    private static final long POLL_INTERVAL_MS = 100L;
+    static long pollIntervalMs = 100L;
 
     private final RemoteLogMetadataSerde serde = new RemoteLogMetadataSerde();
-    private final KafkaConsumer<byte[], byte[]> consumer;
-    private final String metadataTopicName;
+    private final Consumer<byte[], byte[]> consumer;
     private final RemotePartitionMetadataEventHandler remotePartitionMetadataEventHandler;
     private final RemoteLogMetadataTopicPartitioner topicPartitioner;
-    private final Time time;
+    private final Time time = new SystemTime();
 
+    // TODO - Update comments below
     // It indicates whether the closing process has been started or not. If it is set as true,
     // consumer will stop consuming messages, and it will not allow partition assignments to be updated.
-    private volatile boolean closing = false;
-
+    private volatile boolean isClosed = false;
     // It indicates whether the consumer needs to assign the partitions or not. This is set when it is
     // determined that the consumer needs to be assigned with the updated partitions.
-    private volatile boolean assignPartitions = false;
+    private volatile boolean isAssignmentChanged = true;
```

Comment:
I think this comment should also be updated.

Reply:
This is required for the first time the ConsumerThread starts running. If the default value is set to false, the ConsumerTask will be busy executing the while loop (Line 116) without actually doing anything, until some assignment is received. Setting the default value to true prevents it from executing the while loop unnecessarily and makes it wait for an assignment.

Reply:
Maybe let's make a note on this?
e.g. `Flag is set to true to wait for partition assignment on the first execution; otherwise thread will be busy without actually doing anything`

Reply:
Added a note in the comment